### PR TITLE
PAINTROID-206 Updated ScreenShotOnFailRule and added to all UI tests

### PR DESCRIPTION
**Pull request by ldirry**
Since API 29 getExternalStorageDirectory() is deprecated and you are not anymore allowed to directly write or create folders to /storage/emulated/0/. The now recommended function is getExternalFilesDir(String). The new path for the failed screenshots is: /storage/emulated/0/Android/data/org.catrobat.paintroid.test/files/screenshots. Also added the ScreenShotOnFailRule to every espresso ui test.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by StanglTobias** [comment]
Any possibility to locate the screenshots in a place that we can see them on jenkins workspace?

**Review by ThorstenBandel — APPROVED** [review]
LGTM, we should take a look if it is possible to access the pictures  in jenkins as Tobias wrote

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**
